core/server/worker/src/main/java/alluxio/worker/file/DefaultFileSystemWorker.java

```diff
@@ -46,30 +37,25 @@
 import javax.annotation.concurrent.NotThreadSafe;
 
 /**
- * This class is responsible for managing all top level components of the file system worker.
+ * This class is responsible for persisting files when requested by the master and a defunct
+ * {@link FileSystemWorkerClientServiceHandler} which always returns UnsupportedOperation Exception.
  */
 @NotThreadSafe // TODO(jiri): make thread-safe (c.f. ALLUXIO-1624)
-public final class DefaultFileSystemWorker extends AbstractWorker implements FileSystemWorker {
+public final class DefaultFileSystemWorker extends AbstractWorker {
   /** Logic for managing file persistence. */
   private final FileDataManager mFileDataManager;
```

Comment:
Since we need the `FileSystemWorkerServiceHandler` in at least 1.5, I think this class will remain here. I haven't thought enough about where to move the file persistence logic, but I imagine it will go to a separate class (but not `BlockWorker`)